docs/source/reference/storage.rst

```diff
@@ -103,7 +103,25 @@ A cloud storage can be used in :code:`MOUNT` mode, :code:`COPY` mode, or :code:`
     :align: center
     :alt: sky-storage-modes
 
+MOUNT_CACHED mode in detail
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+:code:`MOUNT_CACHED` mode uses `rclone <https://rclone.org/>`_
+to provide a virtual filesystem that is asynchronously synced with the bucket.
+Calling :code:`close()` does not guarantee that the file is written to the bucket.
+rclone will sync written files back to the bucket asynchronously in the order they were written.
+The local filesystem should be fully consistent, but a bucket using 
+`MOUNT_CACHED` on multiple nodes may only be eventually consistent.
+
+Important considerations for :code:`MOUNT_CACHED` mode:
+
+* If files are written faster than they can be uploaded to remote storage, the cache will grow until disk space is exhausted.
+* Files only begin uploading after they are closed by all processes.
+* By default, SkyPilot uses a single transfer at a time to ensure files are committed to remote storage in the same order they are created locally.
+* The write performance depends on the disk tier used for caching - faster disks provide better performance.
+
+In MOUNT_CACHED mode, files only begin uploading after they are closed by all processes. By default, SkyPilot uses transfers the data sequentially to maintain file order of closing.
+When a task using MOUNT_CACHED mode completes, SkyPilot ensures all cached data is successfully uploaded to the remote bucket before marking the task as finished. This guarantees that all task outputs are safely stored in cloud storage, even if the task finished execution before uploads completed. For long-running tasks with frequent writes, this may result in additional time spent flushing the cache after the main computation has finished.
```

Comment:
```suggestion
When a task completes, SkyPilot ensures all cached data is successfully uploaded to the remote bucket before marking the task as finished. This guarantees that all task outputs are safely stored in cloud storage, even if the task finished execution before uploads completed. For long-running tasks with frequent writes, this may result in additional time spent flushing the cache after the main computation has finished.
```